server/list.js

```diff
@@ -320,49 +319,28 @@ function handleUpdates(id, {info: lastInfo, tree: lastTree}) {
   // check all the nodes to see if they have changes
   allPages.forEach((id) => {
     // compare old item to new item
-    const newItem = docsInfo[id]
-    const oldItem = lastInfo[id]
-
-    // since we have a "trash" folder we need to account
-    // for both missing items and "trashed" items
-    const isTrashed = (item) => !item || item.path.split('/')[1] === 'trash'
-    if (!isFirstRun && (isTrashed(newItem) || isTrashed(oldItem))) {
-      const item = isTrashed(oldItem) ? newItem : oldItem
-      const {path, modifiedTime} = item
-      const action = isTrashed(oldItem) ? 'Added' : 'Removed'
-      // FIXME: This does not restore deleted documents which are undone to the same location
-      return cache.purge({
-        url: path,
-        modified: modifiedTime,
-        editEmail: `item${action}`,
-        ignore: ['missing', 'modified']
-      }).catch((err) => {
-        log.debug('Error purging trashed item cache', err)
-      })
-    }
+    const newItem = docsInfo[id] || {}
+    const oldItem = lastInfo[id] || {}
 
-    // don't allow direct purges updates for folders with a home file
-    const hasHome = newItem && (driveBranches[newItem.id] || {}).home
-    if (hasHome) return
+    const newModified = new Date(newItem.modifiedTime)
+    const oldModified = new Date(oldItem.modifiedTime)
+    const hasUpdates = newModified > oldModified
 
-    // if this existed before and the path changed, issue redirects
-    if (oldItem && newItem.path !== oldItem.path) {
-      cache.redirect(oldItem.path, newItem.path, newItem.modifiedTime)
-    } else {
-      // basically we are just calling purge because we don't know the last modified
-      cache.purge({url: newItem.path, modified: newItem.modifiedTime}).catch((err) => {
-        if (!err) return
+    // if no updates reported from drive API, don't purge.
+    if (!hasUpdates) return
 
-        // Duplicate purge errors should be logged at debug level only
-        if (err.message.includes('Same purge id as previous')) return log.debug(`Ignoring duplicate cache purge for ${newItem.path}`, err)
+    cache.purge({id: newItem.id, modified: newItem.modifiedTime}).catch((err) => {
+      if (!err) return
 
-        // Ignore errors if not found or no fresh content, just allow the purge to stop
-        if (err.message.includes('Not found') || err.message.includes('No purge of fresh content')) return
+      // Duplicate purge errors should be logged at debug level only
+      if (err.message.includes('Same purge id as previous')) return log.debug(`Ignoring duplicate cache purge for ${newItem.path}`, err)
```

Comment:
In keeping with some tweaks we've made elsewhere, what do you think about removing the `err` object from the callback? Printing the trace is sometimes confusing to newcomers.